llvm/lib/Target/AArch64/AArch64FrameLowering.cpp

```diff
@@ -2116,9 +2180,11 @@ void AArch64FrameLowering::emitPrologue(MachineFunction &MF,
                       << "\n");
     // Find callee save instructions in frame.
     CalleeSavesBegin = MBBI;
-    assert(IsSVECalleeSave(CalleeSavesBegin) && "Unexpected instruction");
-    while (IsSVECalleeSave(MBBI) && MBBI != MBB.getFirstTerminator())
-      ++MBBI;
+    if (!FPAfterSVECalleeSaves) {
```

Comment:
I think this and the following `if (!FPAfterSVECalleeSaves) ` could be folded into one region like:
https://gist.github.com/MacDue/eff5ffa8cc735c1b56e5987555bf3506 (which also avoids having the incorrect `CalleeSavesBegin` variable when `FPAfterSVECalleeSaves` is active).